lib/blocs/companies_cubit/companies_cubit.dart

```diff
@@ -0,0 +1,66 @@
+import 'dart:async';
+
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:twake/blocs/companies_bloc/companies_bloc.dart';
+import 'package:twake/blocs/notification_bloc/notification_bloc.dart';
+import 'package:twake/blocs/profile_bloc/profile_bloc.dart';
+import 'package:twake/models/globals/globals.dart';
+import 'package:twake/repositories/collection_repository.dart';
+
+class CompaniesCubit extends Cubit<CompaniesState> {
+  final CollectionRepository<Company> repository;
+  final NotificationBloc notificationBloc;
+  late StreamSubscription _notificationSubscription;
+
+  CompaniesCubit(this.repository, this.notificationBloc)
+      : super(CompaniesLoaded(
+          companies: repository.items,
+          selected: repository.selected,
+        )) {
+    ProfileBloc.selectedCompanyId = repository.selected!.id;
+    ProfileBloc.selectedCompany = repository.selected;
+
+    _notificationSubscription =
+        notificationBloc.listen((NotificationState state) {
+      if (state is BaseChannelMessageNotification) {
+        repository.select(state.data.companyId);
+        ProfileBloc.selectedCompanyId = state.data.companyId;
+        final newState = CompaniesLoaded(
+          companies: repository.items,
+          selected: repository.selected,
+        );
+        emit(newState);
+      }
+    });
+  }
+
+  Future<void> fetch() async {
+    await repository.reload();
+    emit(CompaniesLoaded(
+      companies: repository.items,
+      selected: repository.selected,
+    ));
+  }
+
+  Future<void> clear() async {
+    await repository.clean();
+    emit(CompaniesEmpty());
+  }
+
+  void selectCompany(String companyId) {
+    Globals.instance.companyId = companyId;
+    repository.select(companyId);
+    ProfileBloc.selectedCompanyId = companyId;
+    final newState = CompaniesLoaded(
```

Comment:
Also you can review the list of available states for CompaniesCubit, and remove unused ones